src/routes/AddDoctor.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { doctorsCreateThunk } from '../redux/doctor/doctorSlice';
+import { getUser } from '../redux/auth/authSlice';
+
+const AddDoctor = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const fileInputRef = useRef(null);
+
+  const [name, setName] = useState('');
+  const [bio, setBio] = useState('');
+  const [rate, setRate] = useState('');
+  const [specialization, setSpecialization] = useState('');
+  const [image, setImage] = useState('');
+  const [hospital, setHospital] = useState('');
+
+  const user = useSelector(getUser);
+
+  useEffect(() => {
+    if (!user) navigate('/auth');
+  }, [user]);
+
+  const handleFileChange = (event) => {
+    const selectedFile = event.target.files[0];
+    setImage(selectedFile);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const formData = new FormData();
+    formData.append('doctor[name]', name);
+    formData.append('doctor[image]', image);
+    formData.append('doctor[specialization]', specialization);
+    formData.append('doctor[bio]', bio);
+    formData.append('doctor[rate]', rate);
+    formData.append('doctor[hospital]', hospital);
+
+    const userId = user.id;
+    formData.append('doctor[user_id]', userId);
+
+    if (name !== '' && image !== '' && specialization !== '' && bio !== '' && rate !== '' && hospital !== '') {
+      Promise.resolve(dispatch(doctorsCreateThunk(formData))).then(() => {
+        setName('');
+        setImage('');
+        setSpecialization('');
+        setBio('');
+        setRate('');
+        setHospital('');
+
+        if (fileInputRef.current) {
+          fileInputRef.current.value = '';
+        }
+      });
+    }
```

Comment:
[Optional] I think it would be good if the page navigates to the main page after a doctor is added